pkg/manifests/types.go

```diff
@@ -241,6 +243,26 @@ type ThanosQuerierConfig struct {
 	Tolerations []v1.Toleration `json:"tolerations,omitempty"`
 }
 
+// The `NodeExporterConfig` resource defines settings for the Node Exporter component.
+type NodeExporterConfig struct {
+	// Defines collectors enablement and additional parameters.
+	Collectors *NodeExporterCollectorConfig `json:"collectors,omitempty"`
+}
+
+// The `NodeExporterCollectorConfig` resource defines settings for individual collectors
+// of the Node Exporter component.
+type NodeExporterCollectorConfig struct {
+	// cpufreq collector, disabled by default.
+	CpuFreq *NodeExporterCollectorSimpleSwitchConfig `json:"cpufreq,omitempty"`
```

Comment:
I'd keep one struct per collector even though it means repetition for some fields. It will be clearer at least for the documentation IMHO.
It's also easier to use values instead of pointers.

```suggestion
	CpuFreq NodeExporterCpuFreqCollectorConfig `json:"cpufreq,omitempty"`
```

Reply:
thanks for reminding me of the documentation readability, overlooked that point when coding 🙂